scripts/create-references.js

```diff
@@ -11,55 +10,42 @@ const config = JSON.parse(fs.readFileSync('tsconfig.json').toString());
 config.files = [];
 config.references = [];
 
-(async function() {
-
-    const { stdout, stderr } = await exec('yarn workspaces info --json');
-    
-    const lines = stdout.split('\n');
-    const depthTree = lines.slice(1, lines.length - 2).join('\n');
-    let workspaces = undefined;
-
-    // sometimes yarn prints this line: yarn run v1.22.10 and sometimes not
-    try {
-        workspaces = JSON.parse(depthTree);
-    } catch (e) {
-        workspaces = JSON.parse(stdout);
-    }
-
-    for (const name in workspaces) {
-        const workspace = workspaces[name];
-        const location = path.resolve(process.cwd(), workspace.location);
-        const tsconfigPath = path.resolve(location, 'tsconfig.json');
-
-        if (fs.existsSync(tsconfigPath)) {
-            config.references.push({
-                path: workspace.location,
-            });
-            const workspaceConfig = JSON.parse(
-                fs.readFileSync(tsconfigPath).toString(),
-            );
-            workspaceConfig.compilerOptions.composite = true;
-            workspaceConfig.references = [];
-            for (const dependency of workspace.workspaceDependencies) {
-                const dependecyLocation = path.resolve(
-                    process.cwd(),
-                    workspaces[dependency].location,
-                );
-                if (
-                    fs.existsSync(
-                        path.resolve(dependecyLocation, 'tsconfig.json'),
-                    )
-                ) {
-                    workspaceConfig.references.push({
-                        path: path.relative(location, dependecyLocation),
-                    });
-                }
-            }
-            fs.writeFileSync(
-                tsconfigPath,
-                JSON.stringify(workspaceConfig, undefined, 4),
-            );
+(async function () {
+  const { stdout, stderr } = await exec('yarn workspaces info --json');
+
+  const lines = stdout.split('\n');
+  const depthTree = lines.slice(1, lines.length - 2).join('\n');
+  let workspaces = undefined;
+
+  // sometimes yarn prints this line: yarn run v1.22.10 and sometimes not
```

Comment:
But does it work every time? Differing outputs per run can be confusing and is prone to causing more issues as there is no way of confirming whether the command has run properly. Perhaps find out why it has different outputs and write that in the comment as well?

Reply:
If I run this script manually with `node scripts/create-references.js` I get a different output compared to if I add it as a script and run it that way. 

I don't really feel like investigating this any further as its just a trivial developer script. If it doesn't work, you just need to update the references in tsconfig.json manually. It is not really a big deal.